jax/interpreters/xla.py

```diff
@@ -67,14 +67,15 @@
 def compile_or_get_cached(backend, computation, compile_options):
     # Avoid import cycle between jax and jax.experimental
     from jax.experimental.compilation_cache import compilation_cache as cc
-    if cc.is_initialized():
-        cached_executable = cc.get_executable(computation, compile_options)
+    # Persistent compilation cache only implemented on TPU. Avoid errors on CPU.
+    if cc.is_initialized() and backend.platform != 'cpu':
```

Comment:
what about gpu?

can we narrow this to only supported platform ?

Reply:
GPU definitely fails to serialize the executable.

Reply:
I intentionally did it this way so it does raise an error when using GPU, so people know it doesn't work. I think we may eventually wanna intercept all cache exceptions and print a warning but continue running, but for now I think it's better to fail. I could be convinced to just check for TPU though.

Reply:
+ 1 could we do backend.platform == 'tpu' ?

Reply:
Changed to platform == tpu.